src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -56,6 +54,36 @@
                    ["=" ["node" "active"] true]]
                   req))
 
+(defn add-extract [projections query]
```

Comment:
I'm probably going to fork this into another ticket, it might be a fiddly fix, and we've done enough this patch.